app/apptesting/concentrated_liquidity.go

```diff
@@ -6,21 +6,790 @@ import (
+	ExpectedLowerTickSpreadRewardGrowth        sdk.DecCoins
+	ExpectedUpperTickSpreadRewardGrowth        sdk.DecCoins
+	ExpectedSpreadRewardGrowthAccumulatorValue osmomath.Dec
+	// since we use different values for the seondary position's tick, save (tick, expectedSpreadRewardGrowth) tuple
+	ExpectedSecondLowerTickSpreadRewardGrowth SecondConcentratedPosition
+	ExpectedSecondUpperTickSpreadRewardGrowth SecondConcentratedPosition
+
+	NewLowerPrice  osmomath.Dec
+	NewUpperPrice  osmomath.Dec
+	PoolLiqAmount0 osmomath.Int
+	PoolLiqAmount1 osmomath.Int
+	ExpectErr      bool
+}
+
+type SecondConcentratedPosition struct {
+	TickIndex                  int64
+	ExpectedSpreadRewardGrowth sdk.DecCoins
+}
+
 var (
-	ETH                = "eth"
-	USDC               = "usdc"
-	WBTC               = "ibc/D1542AA8762DB13087D8364F3EA6509FD6F009A34F00426AF9E4F9FA85CBBF1F"
-	DefaultTickSpacing = uint64(100)
-	DefaultLowerTick   = int64(30545000)
-	DefaultUpperTick   = int64(31500000)
-	DefaultCoinAmount  = osmomath.NewInt(1000000000000000000)
+	WBTC                       = "ibc/D1542AA8762DB13087D8364F3EA6509FD6F009A34F00426AF9E4F9FA85CBBF1F"
+	DefaultTickSpacing         = uint64(100)
+	DefaultLowerPrice          = osmomath.NewDec(4545)
+	DefaultLowerTick           = int64(30545000)
+	DefaultUpperPrice          = osmomath.NewDec(5500)
+	DefaultUpperTick           = int64(31500000)
+	DefaultCurrPrice           = osmomath.NewDec(5000)
+	DefaultCurrTick      int64 = 31000000
+	DefaultCurrSqrtPrice       = func() osmomath.BigDec {
+		curSqrtPrice, _ := osmomath.MonotonicSqrt(DefaultCurrPrice) // 70.710678118654752440
+		return osmomath.BigDecFromDec(curSqrtPrice)
+	}()
+
+	DefaultSpreadRewardAccumCoins = sdk.NewDecCoins(sdk.NewDecCoin("foo", osmomath.NewInt(50)))
+
+	DefaultCoinAmount = osmomath.NewInt(1000000000000000000)
+
+	// Default tokens and amounts
+	ETH                 = "eth"
+	DefaultAmt0         = osmomath.NewInt(1000000)
+	DefaultAmt0Expected = osmomath.NewInt(998976)
+	DefaultCoin0        = sdk.NewCoin(ETH, DefaultAmt0)
+	USDC                = "usdc"
+	DefaultAmt1         = osmomath.NewInt(5000000000)
+	DefaultAmt1Expected = osmomath.NewInt(5000000000)
+	DefaultCoin1        = sdk.NewCoin(USDC, DefaultAmt1)
+	DefaultCoins        = sdk.NewCoins(DefaultCoin0, DefaultCoin1)
+
+	// Both of the following liquidity values are calculated in x/concentrated-liquidity/python/swap_test.py
+	DefaultLiquidityAmt   = osmomath.MustNewDecFromStr("1517882343.751510417627556287")
+	FullRangeLiquidityAmt = osmomath.MustNewDecFromStr("70710678.118654752941000000")
+
+	swapFundCoins = sdk.NewCoins(sdk.NewCoin("eth", osmomath.NewInt(10000000000000)), sdk.NewCoin("usdc", osmomath.NewInt(1000000000000)))
+
+	roundingError = osmomath.OneInt()
+	EmptyCoins    = sdk.DecCoins(nil)
+
+	// Various sqrt estimates
+	Sqrt4994 = osmomath.MustNewDecFromStr("70.668238976219012614")
+
+	// swap out given in test cases
+	SwapOutGivenInCases = map[string]ConcentratedSwapTest{
```

Comment:
Side-note: one downside of using the map structure is it's not possible to add lists of test cases together when we want to test more general behavior

I wonder if there's a solution to this that doesn't include doing a `name` field which looks a bit messy?